### PR TITLE
Remove a private internal method that is not used.

### LINKED ISSUE #4662: Azure::Core::Operation<T>::GetRawResponseInternal() is never used
This is a pure virtual method. Every derived class needs to implement it, from what I can see, all derived classes have the same implementation with only 1-line code `return *m_rawResponse;`. But this method is not used anywhere, can we just delete it?

I found this problem when working on improving storage test coverage, this line of code can never be reached.

image

### PR DESCRIPTION
**Pull request by RickWinter**
Unused private virtual method

Fixes #4662

### CODE REVIEW

**Review by antkmsft — COMMENTED** [review]
LGTM, but let's wait until after this month release (releasing this would require releasing all libraries), and let's still mention it in the changelog in Breaking Changes section.

**Review by Jinming-Hu — APPROVED** [review]
Changes to storage LGTM

File: sdk/core/azure-core/inc/azure/core/operation.hpp
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/operation.hpp:33]: I believe the intent of this virtual method was to guard against `null` and prevent a seg-fault in derived classes which implement `Operation<T>`. It was added in the following PR by @vhvb1989.
See this test case:
https://github.com/Azure/azure-sdk-for-cpp/pull/1875#discussion_r592868084

I think `GetRawResponse()` needs to call it in `operation.hpp` similar to how `Poll` calls `PollInternal`, and we may have missed that detail in the PR that introduced this method.

Comment (ahsonkhan) [reply sdk/core/azure-core/inc/azure/core/operation.hpp:33]: Looking back at how we ended up here, we had initially designed `Operation<T>` to have a pure **virtual** `GetRawResponse()` method that all derived classes would implement.

The PR that introduced it doesn't mention the design rationale explaining why we want that: https://github.com/Azure/azure-sdk-for-cpp/pull/1803
> Added the pure virtual method

I don't know the reason for that initial design, maybe it was to allow the derived classes to special case how to get the raw response, and maybe do some pre-processing before returning. It looks like we no longer want `GetRawResponse()` to be virtual at all (even via GetRawResponseInternal).

Since this is private inheritance, we can always add that extension point in the future, as needed.

The current `GetRawResponse()` **instance** method does the appropriate null checking, so the seg-fault error doesn't show up anymore.

**I don't see any ownership transfer issues with `m_rawResponse` now, at least not with `GetRawResponse()`, since `Operation<T>` owns it.**

**Comment by antkmsft** [comment]
See also #4749 4749

**Pull request merged**
